### PR TITLE
Add verifyGenericPackageDescription

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/Distribution/PackageDescription/Parsec.hs [hunk 1]
```diff
@@ -522,6 +522,21 @@ When/if we re-implement the parser to support formatting preservging roundtrip
...
+
+-- | This is a post processing step for GPD
+--
+-- Note that if you are unsure if something have to be checked there
+-- or in "Distribution.PackageDescription.Check", it most likely should be there.
+verifyGenericPackageDescription :: GenericPackageDescription -> ParseResult GenericPackageDescription
+verifyGenericPackageDecsription gpd = do
+    -- TODO: currently there aren't any checks, but this is where they can be added.
+    -- Note: if we will check that used flags are the ones declared,
+    -- we can do that already in parseCondTree: then errors will have a correct position.
```
Comment (hvr) [inline Cabal/Distribution/PackageDescription/Parsec.hs]: so... how does one fail here? e.g. let's assume an undeclared flag use-site is detected; how do I throw an error pointing to the use-site of the unknown flag?

**Comment by 23Skidoo** [comment]
Looks like CI failures are due to `-Wall`.

File: Cabal/Distribution/PackageDescription/Parsec.hs (hunk 1 above)
Comment (phadej) [reply Cabal/Distribution/PackageDescription/Parsec.hs]: @hvr with `parseFailure`. As the comment implies, at this point we don't have positions attached, so we can only `parseFailure zeroPos`.

Let's do better flag handling as a follow-up or an alternative.

Comment (phadej) [reply Cabal/Distribution/PackageDescription/Parsec.hs]: Note to previous: Unfortunately we might need to do some trickery-pockery, as flag may be declared after it's usage, if someone puts `flag` stanza after `library`. This is slight :/

**Comment by phadej** [comment]
Cabal-3.0 has `checkForUndefinedFlags`

**Pull request closed without merging**